### PR TITLE
use new lilya

### PR DESCRIPTION
**Pull request by devkral**
### Checklist

- [x] The code has 100% test coverage.
- [x] The documentation was properly created or updated (if applicable) following the correct guidelines and appropriate language.
- [x] I branched out from the latest main or is a sub-branch.

### Summary or description

- Leverage the new lilya encoder API.
- Have encoders per app.
- Use eval_str for newer python versions.

### CODE REVIEW

**Comment by tarsil** [comment]
@devkral no need for this. I already applied the new version. You can close this

File: esmerald/applications.py [hunk 1]
```diff
@@ -1608,7 +1621,15 @@ def extend(self, config: PluggableConfig) -> None:
         ] = State()
         self.async_exit_config = esmerald_settings.async_exit_config
 
-        self.encoders = self.load_settings_value("encoders", encoders) or []
+        self.encoders: list[Union[EncoderProtocol, MoldingProtocol]] = list(
```
Comment (tarsil) [inline esmerald/applications.py:1624]: the typing here is not actually needed. Did mypy complain?

File: esmerald/applications.py [hunk 2]
```diff
@@ -2577,13 +2595,16 @@ def default_settings(
         """
         return esmerald_settings
 
-    async def globalise_settings(self) -> None:
+    async def globalize_settings(self) -> None:
```
Comment (tarsil) [inline esmerald/applications.py:2598]: globalise with an S, I wrote in british english

File: esmerald/applications.py [hunk 3]
```diff
@@ -2577,13 +2595,16 @@ def default_settings(
...
         """
         Making sure the global settings remain as is
         after the request is done.
         """
         esmerald_settings.configure(__lazy_settings__._wrapped)
 
+    # typo
+    globalise_settings = globalize_settings
```
Comment (tarsil) [inline esmerald/applications.py:2606]: rollback to S and remove this

File: esmerald/applications.py [hunk 4]
```diff
@@ -2594,7 +2615,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
 
         scope["state"] = {}
         await super().__call__(scope, receive, send)
-        await self.globalise_settings()
+        await self.globalize_settings()
```
Comment (tarsil) [inline esmerald/applications.py:2618]: I won't be making all comments about this but rollback, please

File: esmerald/encoders.py [hunk 5]
```diff
@@ -1,102 +1,70 @@
...
 
-import msgspec
-from lilya._internal._encoders import json_encoder as json_encoder  # noqa
-from lilya._utils import is_class_and_subclass
+from lilya._internal._encoders import ModelDumpEncoder
 from lilya.encoders import (
-    ENCODER_TYPES as LILYA_ENCODER_TYPES,  # noqa
-    Encoder as LilyaEncoder,  # noqa
-    register_encoder as register_encoder,  # noqa
+    ENCODER_TYPES as ENCODER_TYPES_CTX,  # noqa
```
Comment (tarsil) [inline esmerald/encoders.py:7]: LILYA_ENCODER_TYPES was ok, no need to rename, please rollback

File: esmerald/encoders.py [hunk 6]
```diff
@@ -1,102 +1,70 @@
...
+        register_encoder(encoder)
+    except RuntimeError:
+        raise ImproperlyConfigured(f"{type(encoder)} must be a subclass of Encoder") from None
 
 
-class PydanticEncoder(Encoder):
-    def is_type(self, value: Any) -> bool:
-        return isinstance(value, BaseModel) or is_class_and_subclass(value, BaseModel)
+PydanticEncoder = ModelDumpEncoder
+try:
```
Comment (tarsil) [inline esmerald/encoders.py:34]: NO need to use the try here. Msgspec is a hard dependency for now. Rollback this

File: esmerald/encoders.py [hunk 7]
```diff
@@ -1,102 +1,70 @@
...
-    if not isinstance(encoder, Encoder) and not is_class_and_subclass(encoder, Encoder):  # type: ignore
-        raise ImproperlyConfigured(f"{type(encoder)} must be a subclass of Encoder")
+    register_esmerald_encoder(MsgSpecEncoder)
+except ImportError:
+    pass
 
-    encoder_types = {encoder.__class__.__name__ for encoder in ENCODER_TYPES}
-    if encoder.__name__ not in encoder_types:
-        register_encoder(encoder)
+json_encoder = json_encode
```
Comment (tarsil) [inline esmerald/encoders.py:54]: I changed everything to use json_encode already, no need to use this. Why? Rollback please

File: esmerald/encoders.py [hunk 8]
```diff
@@ -1,102 +1,70 @@
...
 
 
 def is_body_encoder(value: Any) -> bool:
     """
     Function that checks if the value is a body encoder.
     """
+    encoder_types = ENCODER_TYPES_CTX.get()
     if not is_union(value):
-        return any(encoder.is_type(value) for encoder in ENCODER_TYPES)
+        return any(encoder.is_type(value) for encoder in encoder_types)
```
Comment (tarsil) [inline esmerald/encoders.py:63]: I want this like it was before. ENCODER_TYPES = LILYA_ENCODER_TYPES.get() at the top and its used here directly. No need for reassignments, at all.

File: esmerald/responses/base.py [hunk 9]
```diff
@@ -23,10 +25,10 @@
     Response as LilyaResponse,  # noqa
     StreamingResponse as StreamingResponse,  # noqa
 )
-from orjson import OPT_OMIT_MICROSECONDS, OPT_SERIALIZE_NUMPY, dumps
+from orjson import OPT_OMIT_MICROSECONDS, OPT_SERIALIZE_NUMPY, dumps, loads
 from typing_extensions import Annotated, Doc
 
-from esmerald.encoders import Encoder, json_encoder
+from esmerald.encoders import ENCODER_TYPES_CTX, Encoder, json_encoder
```
Comment (tarsil) [inline esmerald/responses/base.py:31]: Once LILYA_ENCODER_TYPES is rolledback, this should be changed.

Literally, why change to json_encoder when the base already was updated to use the json_encode? Rollback all of this.

File: esmerald/responses/encoders.py [hunk 10]
```diff
@@ -17,15 +20,48 @@ class ORJSONResponse(BaseJSONResponse):
     In the same way the JSONResponse is used, so is the `ORJSONResponse`.
     """
 
+    @staticmethod
```
Comment (tarsil) [inline esmerald/responses/encoders.py:23]: Why is this here? The transform should be as it was before. In the Response which is the base class of all of them. This is over complicating. No reason

File: esmerald/responses/encoders.py [hunk 11]
```diff
@@ -17,15 +20,48 @@ class ORJSONResponse(BaseJSONResponse):
...
+            )
+            if self.encoders
+            else None
+        )
+        return cast(
+            bytes,
+            json_encode(
+                content,
+                json_encode_fn=partial(
+                    orjson.dumps, option=orjson.OPT_SERIALIZE_NUMPY | orjson.OPT_OMIT_MICROSECONDS
```
Comment (tarsil) [inline esmerald/responses/encoders.py:56]: This is ok to override since you want orjson here but the transform should be at the base as it was before.

File: esmerald/transformers/signature.py [hunk 12]
```diff
@@ -19,7 +20,7 @@
 from orjson import loads
 from pydantic import ValidationError, create_model
 
-from esmerald.encoders import ENCODER_TYPES, Encoder
+from esmerald.encoders import ENCODER_TYPES_CTX, Encoder
```
Comment (tarsil) [inline esmerald/transformers/signature.py:23]: Again , rollback to what it was, Literally, please

File: esmerald/transformers/signature.py [hunk 13]
```diff
@@ -516,12 +521,19 @@ def _find_encoder(self, annotation: Any) -> Any:
             Any: The encoder found, or None if no encoder matches.
         """
         origin = get_origin(annotation)
-        for encoder in ENCODER_TYPES:
-            if not origin and encoder.is_type(annotation):
+        for encoder in ENCODER_TYPES_CTX.get():
+            if (
+                not origin
+                and hasattr(encoder, "is_type_structure")
+                and encoder.is_type_structure(annotation)
```
Comment (tarsil) [inline esmerald/transformers/signature.py:528]: The reason why all the tests are failing its because you removed the custom encoders of Esmerald. They differ a bit from Lilya, intentionally. 

Roll everything back to the way it was before. There is a reason why its using the `is_type` and not the structure. It was ALL working before.

This was not needed whatsoever

**Review by tarsil — CHANGES_REQUESTED** [review]

File: esmerald/transformers/signature.py (hunk 13 above)
Comment (devkral) [reply esmerald/transformers/signature.py:528]: yep, this will work, despite a bit hacky (subclass check in is_type is a bit unexpected), I think I will undo most

**Comment by devkral** [comment]
This PR is too invasive. I close it and do it more smart

File: esmerald/transformers/signature.py (hunk 13 above)
Comment (tarsil) [reply esmerald/transformers/signature.py:528]: Rollback everything. Esmerald is not Lilya and it operates differently. Not a hack

**Comment by tarsil** [comment]
Don't change things that are already working.
